src/lib/packet.ts

```diff
@@ -8,16 +7,40 @@ export class ServerPacket {
         this.data = data;
     }
 
+    getType() : string {
+        return this.type
+    }
+
+    getData() : unknown {
+        return this.data
+    }
+
     toString() : string {
```

Comment:
This is not Java, a toString function is not needed.
You can just call JSON.stringify on the object and get the same result.
e.g. JSON.stringify(this)